### ISSUE TITLE
x509: certificate signed by unknown authority

### ISSUE THREAD

**Issue by HuKeping** [docker/notary#816]
Hi,

I deploy notary service on a machine(host1) and try `docker pull xxx` from another(host2) with env
```
export DOCKER_CONTENT_TRUST=1
export DOCKER_CONTENT_TRUST_SERVER=https://notary-server:4443
```

From host2, I get 'x509: certificate signed by unknown authority'
From host1, per the log of notary-server I get `http: TLS handshake error from IP:Port: remote error: bad certificate`

It was strange that even if I installed the root-ca into the system, I can still get those error.

And then I check the code for `trust pull` in docker/docker, https://github.com/docker/docker/blob/master/api/client/trust.go#L139

It indicates that it will try to load the certs from ~/.docker/tls/notary-server:4443/ so I put the root-ca into that directory, finally it works.

**Comment by HuKeping** [comment]
The OS of host2 is centos and I install the CA by
```
$ sudo cp root-ca /etc/pki/ca-trust/source/anchors
$ sudo update-ca-trust extract
```


I also try the same operation of host2 on host3(Ubuntu). First I get the x509 error which was as expected because I haven't install the CA into system yet. But after I installed the CA, it works fine.
```
$ sudo mkdir /usr/share/ca-certificates/notary
$ sudo cp root-ca /usr/share/ca-certificates/notary/
$ sudo dpkg-reconfigure ca-certificates
```

So why it still get the error `'x509: certificate signed by unknown authority`  even if I have installed the root-ca into system on host2.

Do we always need to put root-ca into directory like `~/.docker/tls/notary-server:4443/ `?
